Backend/app/main.py

```
@@ -32,9 +33,28 @@ async def create_tables():
 @asynccontextmanager
 async def lifespan(app: FastAPI):
     print("App is starting...")
+    # Initialize Supabase async client (non-fatal if missing envs)
+    supabase_url = os.getenv("SUPABASE_URL")
+    supabase_key = os.getenv("SUPABASE_KEY")
+    if supabase_url and supabase_key:
+        try:
+            app.state.supabase = create_client(supabase_url, supabase_key, supabase_cls=AsyncClient)
+        except Exception as exc:  # broad to avoid startup crash; logs include stack
+            app.state.supabase = None
+            logging.exception("Failed to initialize Supabase AsyncClient: %s", exc)
```

Ruff (0.14.10)

Redundant exception object included in `logging.exception` call

(TRY401)